contracts/RiskFund/RiskFund.sol

```diff
@@ -141,16 +141,18 @@ contract RiskFund is Ownable2StepUpgradeable, ExponentialNoError, ReserveHelpers
         );
         require(canSwapPoolsAsset, "Risk fund: Not authorized to swap pool assets.");
         require(poolRegistry != address(0), "Risk fund: Invalid pool registry.");
-        require(
-            underlyingAssets.length == amountsOutMin.length,
-            "Risk fund: underlyingAssets and amountsOutMin are unequal lengths"
-        );
+        require(markets.length == amountsOutMin.length, "Risk fund: markets and amountsOutMin are unequal lengths");
 
         uint256 totalAmount;
-        uint256 underlyingAssetsCount = underlyingAssets.length;
-        for (uint256 i; i < underlyingAssetsCount; ++i) {
-            VToken vToken = VToken(underlyingAssets[i]);
+        uint256 marketsCount = markets.length;
+        for (uint256 i; i < marketsCount; ++i) {
+            VToken vToken = VToken(markets[i]);
             address comptroller = address(vToken.comptroller());
+
+            PoolRegistry.VenusPool memory pool = PoolRegistry(poolRegistry).getPoolByComptroller(comptroller);
+            require(pool.comptroller == comptroller, "comptroller doesn't exist pool registry");
```

Comment:
Here we are getting pool through comptroller and then verifing the pool's comptroller address to comptroller address. Is this require necessary here?

Reply:
`getPoolByComptroller` doesn't check if the comptroller is registered. So for any address it returns a `pool` with 0 values. So this is the way to check if the comptroller is registered.